src/bin/common/solving_loop.ml

```diff
@@ -886,22 +886,32 @@ let main () =
 
       | {contents = `Get_assignment; _} ->
         begin
-          match State.get partial_model_key st with
-          | Some Model ((module SAT), partial_model) ->
-            if State.get produce_assignment st then
-              handle_get_assignment
-                ~get_value:(SAT.get_value partial_model)
-                st
-            else
+          if Options.get_interpretation () then begin
+            match State.get partial_model_key st with
+            | Some Model ((module SAT), partial_model) ->
+              if State.get produce_assignment st then
+                handle_get_assignment
+                  ~get_value:(SAT.get_value partial_model)
+                  st
+              else
+                recoverable_error
+                  "Produce assignments disabled; \
+                   add (set-option :produce-assignments true)";
+              st
+            | None ->
+              (* TODO: add the location of the statement. *)
               recoverable_error
-                "Produce assignments disabled; \
-                 add (set-option :produce-assignments true)";
-            st
-          | None ->
-            (* TODO: add the location of the statement. *)
-            recoverable_error
-              "No model produced, cannot execute get-assignment.";
-            st
+                "No model produced, cannot execute get-assignment.";
+              st
+          end
+          else
+            begin
+              (* TODO: add the location of the statement. *)
+              recoverable_error
+                "(get-assignment) requires model generation, \
+                 but it is disabled (try --produce-models)";
+              st
+            end
```

Comment:
I thought we determined earlier that `produce-assignments` should work without `produce-models`, why are we disabling it instead?

Reply:
Yes, I wanted a successful test :) Indeed the `sat` does not require to have built the model explicitly.

> Because there already was a test without (set-option :produce-models true).

Sorry, I must have missed it, but I still don't see it — can you point me to that test? The tests I see are `bool1`, `bool2`, `bool3` that have both `produce-models` and `produce-assignments`; `get-assignment1` that has neither, `get-assignment2` that has `produce-models` but not `produce-assignments` and only the new `get-assignment3` that has `produce-assignment` but not `produce-models`.

Reply:
> We should have a test without (set-option :produce-models true), otherwise looks good (modulo the integer / boolean typo).
 
`tests/smtlib2/testfile-get-assignment1.err.dolmen.smt2` indeed has neither

Reply:
Ah! Sorry I was not clear, I meant without `produce-models` but with `produce-assignments` as clarified in my later message :)